src/webui/src/components/trial-detail/Para.tsx

```diff
@@ -79,23 +83,30 @@ class Para extends React.Component<ParaProps, ParaState> {
             lengthofTrials: number
         ): void => {
             // get data for every lines. if dim is choice type, number -> toString()
-            const paraYdata: number[][] = [];
-            Object.keys(eachTrialParams).map(item => {
-                const temp: number[] = [];
-                for (let i = 0; i < dimName.length; i++) {
-                    if ('type' in parallelAxis[i]) {
-                        temp.push(eachTrialParams[item][dimName[i]].toString());
-                    } else {
-                        // default metric
-                        temp.push(eachTrialParams[item][dimName[i]]);
+            let paraYdata: number[][] = [];
+            const { isNested } = this.state;
+            if (isNested === false) {
+                Object.keys(eachTrialParams).map(item => {
```

Comment:
I think `for ... of` loop should be prefered if the return value is not used.

Reply:
Agreed. foreach(item => ... is also better